chore: remove dependency on thecodingmachine/cache-utils

xyng opened the pull request:
The package seems to be unmaintained and prevents usage due to old `psr/simple-cache`.
This removes the dependency on that package, which provided cache clearance on Classes and ParentClasses when their files changed.
The functionality can still be achieved by creating a custom implementation of `TheCodingMachine\GraphQLite\Utils\Cache\ClassBoundCacheContractFactoryInterface` that returns a wrapped version of `thecodingmachine/cache-utils` or an updated fork.

My implementation is quite naive and changes how the caching works. This may be considered a breaking change.

I'm happy to make adjustments if needed.

fixes thecodingmachine/graphqlite#693

Note: PHPStan seems to have problems. This is probably not related to my changes.

xyng committed:
chore: remove dependency on thecodingmachine/cache-utils

The package seems to be unmaintained and prevents usage due to old
psr/simple-cache. This removes the dependency on that package.
The package provided cache clearance on Classes and ParentClasses when their files changed.
This functionality can still be achieved by creating a custom implementation of
`TheCodingMachine\GraphQLite\Utils\Cache\ClassBoundCacheContractFactoryInterface`.

fixes thecodingmachine/graphqlite#693

andrew-demb commented on src/FactoryContext.php:
```diff
@@ -28,6 +29,7 @@ public function __construct(
         private readonly RecursiveTypeMapperInterface $recursiveTypeMapper,
         private readonly ContainerInterface $container,
         private readonly CacheInterface $cache,
+        private readonly ClassBoundCacheContractFactoryInterface|null $classBoundCacheContractFactory = null,
```
Shoud it be added at the end of the list to preserve BC?

xyng replied:
Good Catch! I added it near to cache for no particular reason (maybe optics?).
Will adjust later.

oojacoboo commented:
@xyng thanks for this.  For PHPStan, yea, not sure what's going on with that.  It looks like a syntax recommended change.  If you could give that a shot, that'd be great.

I think it'd be best to place this in `src/Cache` and not in the `src/Utils/Cache` directory.  `Utils` directories are like junk drawers, the mere existence of them just encourages more organization.

I think with @andrew-demb's recommendation on the `__constructor` ordering, there shouldn't be any BC issues.

xyng commented:
@oojacoboo @andrew-demb i have adjusted according to your feedback. Thanks a lot!
I also added a setter `setClassBoundCacheContractFactory` to the `SchemaFactory` to actually allow users to customize.

xyng committed:
fix: correctly escape `|` in phpstan ignoreErrors

xyng committed:
fix: sync min symfony/cache version of library to example and solve deprecation

This is necessary because symfony/cache is not compatible with every supported
version of psr/simple-cache.
More info here: https://github.com/symfony/symfony/issues/44738
